File: CS/WindowsApplication9/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraGrid.Views.Grid;
using System.Collections;

namespace WindowsApplication9
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();
        }

        private void Form1_Load(object sender, EventArgs e)
        {
            for (int i = 0; i < 20; i++) {
                dataTable1.Rows.Add(new object[] { "Product"+ i,i, i%3});
            }
        }

        private void gridView1_CustomColumnSort(object sender, DevExpress.XtraGrid.Views.Base.CustomColumnSortEventArgs e)
        {
				GridView view = sender as GridView;
                BindingSource ds = view.DataSource as BindingSource;
                int category1 = (int)((DataRowView)ds[e.ListSourceRowIndex1])["CategoryID"];
                int category2 = (int)((DataRowView)ds[e.ListSourceRowIndex2])["CategoryID"];
				e.Handled = true;
                if (category1 > category2 && e.SortOrder == DevExpress.Data.ColumnSortOrder.Ascending
                    || category1 < category2 && e.SortOrder == DevExpress.Data.ColumnSortOrder.Descending)
					e.Result = 1;
				else
                    if (category1 > category2 && e.SortOrder == DevExpress.Data.ColumnSortOrder.Descending 
                        || category1 < category2 && e.SortOrder == DevExpress.Data.ColumnSortOrder.Ascending)
                        e.Result = -1;
                    else
                    {
                        e.Result = Comparer.Default.Compare(e.Value1, e.Value2 );
                    }
        }
    }
}

File: VB/WindowsApplication9/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraGrid.Views.Grid
Imports System.Collections

Namespace WindowsApplication9
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()
		End Sub

		Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
			For i As Integer = 0 To 19
				dataTable1.Rows.Add(New Object() { "Product" & i,i, i Mod 3})
			Next i
		End Sub

		Private Sub gridView1_CustomColumnSort(ByVal sender As Object, ByVal e As DevExpress.XtraGrid.Views.Base.CustomColumnSortEventArgs) Handles gridView1.CustomColumnSort
				Dim view As GridView = TryCast(sender, GridView)
				Dim ds As BindingSource = TryCast(view.DataSource, BindingSource)
				Dim category1 As Integer = CInt(Fix((CType(ds(e.ListSourceRowIndex1), DataRowView))("CategoryID")))
				Dim category2 As Integer = CInt(Fix((CType(ds(e.ListSourceRowIndex2), DataRowView))("CategoryID")))
				e.Handled = True
				If category1 > category2 AndAlso e.SortOrder = DevExpress.Data.ColumnSortOrder.Ascending OrElse category1 < category2 AndAlso e.SortOrder = DevExpress.Data.ColumnSortOrder.Descending Then
					e.Result = 1
				Else
					If category1 > category2 AndAlso e.SortOrder = DevExpress.Data.ColumnSortOrder.Descending OrElse category1 < category2 AndAlso e.SortOrder = DevExpress.Data.ColumnSortOrder.Ascending Then
						e.Result = -1
					Else
						e.Result = Comparer.Default.Compare(e.Value1, e.Value2)
					End If
				End If
		End Sub
	End Class
End Namespace